radix-engine/src/blueprints/resource/non_fungible/non_fungible_proof.rs

```diff
@@ -156,18 +158,25 @@ impl NonFungibleProofBlueprint {
 
     pub(crate) fn drop<Y>(proof: Proof, api: &mut Y) -> Result<(), RuntimeError>
     where
-        Y: ClientApi<RuntimeError>,
+        Y: KernelSubstateApi<SystemLockData> + ClientApi<RuntimeError>,
     {
-        let parent = api
-            .get_object_info(proof.0.as_node_id())?
-            .outer_object
-            .unwrap();
-
-        api.call_method(
-            parent.as_node_id(),
-            RESOURCE_MANAGER_DROP_PROOF_IDENT,
-            scrypto_encode(&ResourceManagerDropProofInput { proof }).unwrap(),
+        // TODO: add `drop` callback for drop atomicity, which will remove the necessity of kernel api.
+
+        // Notify underlying buckets/vaults
```

Comment:
Also here

Reply:
Not here. If we drop, we lose visibility to the buckets/vaults.